File: content/examples/other/leetcode/problems/missing-number/example-2/solution.js
function missingNumber(nums) {
  let x = 0;
  for (let i = 0; i <= nums.length; i += 1) x ^= i;
  for (const v of nums) x ^= v;
  return x;
}

if (require.main === module) {
  console.log(missingNumber([3, 0, 1]));
}

module.exports = { missingNumber };

File: content/examples/other/leetcode/problems/missing-number/example-2/Solution.java
public class Solution {
  static int missingNumber(int[] nums) {
    int x = 0;
    for (int i = 0; i <= nums.length; i += 1) x ^= i;
    for (int v : nums) x ^= v;
    return x;
  }

  public static void main(String[] args) {
    System.out.println(missingNumber(new int[] {3, 0, 1}));
  }
}
